test/src/main/java/org/axonframework/test/fixture/AxonTestPhase.java

```diff
@@ -495,23 +448,13 @@ interface Message<T extends Message<T>> {
              */
             T events(@Nonnull EventMessage<?>... expectedEvents);
 
-            /**
-             * Expect the published events during the {@link When} phase to match the given {@code matcher}.
-             * <p>
-             * Note: if no events were published, the matcher receives an empty List.
-             *
-             * @param matcher The matcher to match with the actually published events.
-             * @return The current Then instance, for fluent interfacing.
-             */
-            T events(@Nonnull Matcher<? extends List<? super EventMessage<?>>> matcher);
-
             /**
              * Allow to consume the set of event messages which have been published during the "when" phase.
              *
              * @param consumer Consumes the published events. You may place your own assertions here.
              * @return The current Then instance, for fluent interfacing.
              */
-            T events(@Nonnull Consumer<List<? super EventMessage<?>>> consumer);
```

Comment:
IDE hints are better in this case, before default type here was "Object" whie using lambda notation without specifying the parameter type. I think it's the most common use case. If someone really want inherited type, they may cast inside.